app/views/layouts/navigation.html.erb

```diff
@@ -0,0 +1,31 @@
+<div class="container-fluid">
+  <div class="row">
+    <div class="col-xs-12 col-sm-1 no-horiz-padding">
+      <ul class="nav nav-pills nav-stacked">
+        <% HqController::NAVIGATION_BUTTONS.each do | _, btn | %>
+          <li role="presentation" <%= (request.path=~ btn[:path_regex]) ? 'class=active' : 'class=dormant' %>>
+            <%= link_to %Q{
+              <span class="glyphicon #{btn[:glyph]}" aria-hidden="true"></span>
+              <p class="text-center">#{btn[:text]}</p>
+              }.html_safe, eval(btn[:href]) %>
```

Comment:
Yes, it is weird they're on the same line, but the evaluations are being done on constants, and as such, should not contain active content.  But `eval` is scary.